### ISSUE TITLE
Error Code=-2000 "Loading page failed" is received after logout and login again with Google.

### ISSUE THREAD

**Issue by lucianboboc** [aws-amplify/aws-sdk-ios#1001]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?
AWSCognitoAuth and Google federated authentication

* In what version of SDK are you facing the problem?
iOS 11

* Is the issue limited to Simulators / Actual Devices?
no

* Can your problem be resolved if you bump to a higher version of SDK?
no

* Is this problem related to specific iOS version?
no

* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
using Cocoapods

* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.
Step 1. Login with google
Step 2. Sign out
Step 3. Login again with google

The problem happens just once, if you login with google, then logout and try to log back in, the error is received it works. (`Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-2000 "(null)" UserInfo={error=Loading page failed}`).
If the user is already logged in with Google in the browser and Cognito will not redirect the page to the google login page, when `AWSCognitoAuth's application:openURL:` method is called, `code` and `state` parameters are available and the `/oauth2/token` request is performed, session will be received, saved and available the next time another login call is performed.

**Comment by pothunoorisk** [comment]
@lucianboboc  same issue. Were you able to resolve this?

**Comment by behrooziAWS** [comment]
I haven't been able to get to this yet, it's still on my plate to investigate.

**Comment by anupsarode** [comment]
I too encounter this issue frequently.

In my iOS app, I have a facebook signin button. I call the CognitoAuth to get authenticated and it do get the session object that has id token. But only after clicking the signin button twice. The first call always fails and gives the error

Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-2000 "(null)" UserInfo={error=Loading page failed}

Any solution is appreciated.

Best,
Anup

**Comment by rohandubal** [comment]
Hello @anupsarode @lucianboboc @pothunoorisk 

We have released SDK version `2.7.2` which addresses this issue.

Please update the SDK and let us know if you continue experiencing issues.

Thanks,
Rohan

**Comment by omerwaqaskhan** [comment]
We are facing the same issue in latest SDK

**Comment by muellerfr** [comment]
Hi @omerwaqaskhan,

Can you please confirm which  version of the SDK you are using?

**Comment by rohandubal** [comment]
Hello @omerwaqaskhan 

Could you please specify if you specifically face this issue with Google/ Facebook? Can you also specify the simulator/ device iOS  version and model?

I have written a sample application where I was able to reproduce this issue with SDK version < 2.7.2 but in the latest SDK version it is working as expected. So some input from you will be helpful in identifying a fix.

Thanks,
Rohan

**Comment by rohandubal** [comment]
@anupsarode @lucianboboc @pothunoorisk would be helpful if you can share your experience using this feature in the latest SDK version (`2.7.3`+)

**Comment by lucianboboc** [comment]
@rohandubal for me v2.7.2 works fine. I'm able to logout and login again with google without any error.

**Comment by rohandubal** [comment]
That's great to hear @lucianboboc! 

I will keep this issue open for a couple more days to make sure no one else is facing this issue.

Thanks,
Rohan

**Comment by BillBunting** [comment]
After upgrading to 2.8.1 and switching to using:
AWSMobileClient.sharedInstance().initialize { (userState, error) in
...
AWSMobileClient.sharedInstance().showSignIn(navigationController: nav,signInUIOptions: SignInUIOptions(...

When the "Sign in with Google" button is selected, the login screen goes away and the callback is never called and the user is not signed in.  

Is this the same issue or a new or different issue.  Any ideas?

**Comment by palpatim** [comment]
@BillBunting Sorry for the delayed response on this. If you are still experiencing this on the latest SDK, please open a new issue--we don't believe it would be related to the same root cause.
